### PR TITLE
add a test for the fix

### PR DESCRIPTION
**Pull request by maquessime**
not sure if the strategy is one `.yml` per test case

Depends on #268

### CODE REVIEW

**Comment by WaleedAshraf** [comment]
Nice.
So, if we base this change on current master, CI would fail. Right?

About adding one .yaml per test.
I think yes, we should have a test suie with different yaml files and cases. But that's a separate task.

**Comment by maquessime** [comment]
> So, if we base this change on current master, CI would fail. Right?
correct - that's why I did it against your fix branch 😄

File: examples/2.0.0/anyof.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ components:
       properties:
         key:
           type: string
+          additionalProperties: false
```
Comment (fmvilas) [inline examples/2.0.0/anyof.yml]:
```suggestion
      additionalProperties: false
```

**Comment by fmvilas** [comment]
Oh, just saw your reply. Forget my comment then. You were pointing to his branch. Let me point to master so we don't have branches that depend on other branches.

**Comment by WaleedAshraf** [comment]
let me re-run tests, just to confirm.
it should fail on the master.

**Comment by maquessime** [comment]
to double check I did

- `git revert 4032d954766689693d6d5fd34f6fed975170e5d7`
- `npm run test`
- got 
```
Error details:

[ { keyword: 'type',
    dataPath:
     '.components.schemas[\'objectWithKey\'].properties[\'key\'].additionalProperties',
    schemaPath: '#/allOf/1/type',
    params: { type: 'object' },
    message: 'should be object' } ]
```
- `git cherry-pick 4032d954766689693d6d5fd34f6fed975170e5d7`
- tests were passing

**Comment by WaleedAshraf** [comment]
Eh, you have created a branch from mine. Makes sense why they are still passing.

**Comment by WaleedAshraf** [comment]
So, now we only need to merge this. I can close my PR.

**Comment by maquessime** [comment]
Yeah sorry for the confusion!

**Pull request merged**